Frameworks/Plugins/ZappGoogleInteractiveMediaAds/Files/Universal/GoogleInteractiveMediaAdsAdapter/GoogleInteractiveMediaAdsAdapter.swift

```diff
@@ -69,16 +72,55 @@ import ZappCore
     }
 
     func prepareGoogleIMA() {
+        isPlaybackPaused = false
+        isVMAPAdsCompleted = false
         guard let player = avPlayer else { return }
+        addNotificationsObserver()
+        addRateObserver()
         urlTagData = GoogleUrlTagData(entry: playerPlugin?.entry,
                                       pluginParams: configurationJSON)
         contentPlayhead = IMAAVPlayerContentPlayhead(avPlayer: player)
 
         if let urlToPresent = urlTagData?.prerollUrlString() {
             requestAd(adUrl: urlToPresent)
         }
+        
     }
 
+    func addNotificationsObserver() {
+        let defaultCenter = NotificationCenter.default
+        
+        defaultCenter.addObserver(self,
+                                  selector: #selector(self.applicationWillResignActive(notification:)),
+                                  name: UIApplication.willResignActiveNotification,
+                                  object: nil)
+        
+        defaultCenter.addObserver(self,
+                                  selector: #selector(self.applicationDidBecomeActive(notification:)),
+                                  name: UIApplication.didBecomeActiveNotification,
+                                  object: nil)
+    }
+    
+    @objc func applicationWillResignActive(notification:Notification) {
+        adsManager?.pause()
+    }
+    
+    @objc func applicationDidBecomeActive(notification:Notification) {
+        adsManager?.resume()
+    }
+
+    func addRateObserver() {
+        avPlayer?.addObserver(self, forKeyPath: "rate", options: NSKeyValueObservingOptions.new, context: nil)
```

Comment:
Please more "rate" to constant

Reply:
@kononenkoAnton Thanks for the review. PR updated.